de1plus/plugins/web_api/plugin.tcl

```diff
@@ -77,5 +78,25 @@ proc ::wibble::togglePowerOff {state} {
     dict set response status 200
     dict set state response header content-type "" {application/json charset utf-8}
     dict set response content "{status: \"ok\"}"
+    sendresponse $response
+}
+
+proc ::wibble::checkStatus {state} {
+    if { ![check_auth $state] } {
+        return;
+    }
+
+    dict set response status 200
+    dict set state response header content-type "" {text/plain charset utf-8}
+    
+    # Returning a simple text 1 if the machine is in anything other than an idle state. Return text 0 if idle.
```

Comment:
Might want to update the comment here to reflect the logic below

Reply:
Made a subsequent update to change the comment to refer to sleep states rather than idle states, good catch.